### ISSUE TITLE
Workshop at KTH - July 3

### ISSUE THREAD

**Issue by bbaudry** [KTH/ci-hackathon#10]
We will prepare templates, in different languages, to facilitate the setup for hackathon participants. templates should include all code and instructions that are necessary to connect to the Travis websocket and to the external libraries for visualization and sonification. These templates will be in a [dedicated folder](https://github.com/KTH/ci-hackathon/tree/master/templates).

We will let hackathon participants connect to two hardware fixtures that are unique to R1. For this we need
- a web server that exposes a REST API for the MIDI controller.
- a web server that exposes a REST API for the DMX controller
- each server will exist in two versions: 
  - one that is connect to an emulator / simulator of the hardware: code goes in the [emulators](https://github.com/KTH/ci-hackathon/tree/master/emulators) folder. These servers should run on the local machine of participants in order to let everyone experiment / test.
  - one that is connected to the actual hardware controller: code goes in the [hardware](https://github.com/KTH/ci-hackathon/tree/master/hardware) folder. These servers will run on Raspberry PI connected to the HW controllers. These servers should have a minimalistic control over the origin and the number of connections.

@SimoneStefani buys the ciart.live domain. We will use this domain for the different servers used during the hackathon.
@SimoneStefani is the main responsible to build the MIDI API and servers
@Jacarte and @christian-stj will design the DMX API and servers
We use the issue tracker of this repo to discuss the different steps for the construction of these APIs. We continue using email as well for global discussions.

**Comment by SimoneStefani** [comment]
`ciart.live` bought
Ping me if you want to have a subdomain pointing to some IP

**Comment by oscarlvp** [comment]
Why not use Github templates? See here: https://help.github.com/en/articles/creating-a-template-repository

**Comment by SimoneStefani** [comment]
Yes we discussed yesterday that it can be a nice solution. I think when we are a little bit further in the preparation we can cleanup the repo and make it a template

**Comment by tdurieux** [comment]
> ciart.live bought

If you want to point the hostname to my server, you can do add CNAME dns entry to ce8b7f44-5295-4335-841b-f8b0d7a05576.pub.cloud.scaleway.com

If you do it, tell me which subdomain you used, I need to change my ngnix config. 
Unfortunately, it will not be HTTPS, I don't succeed to configure ngnix to support HTTPS over websocket, on my server, it is Cloudflare that handles the HTTPS.

**Comment by Jacarte** [comment]
Hi @SimoneStefani, could your company host the emulator for [lights](https://github.com/KTH/ci-hackathon/tree/master/resources/emulators/lights)?

**Comment by bbaudry** [comment]
@SimoneStefani will setup lights.ciart.live to host the lights simulator on one of his company's machines

thanks!

**Comment by Jacarte** [comment]
Hi, @SimoneStefani could deploy the new version of the emulator?  Thanks !
